02_bubble/starfallz/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
```

Comment:
the function should not modify passed slice:
`Write a function that returns a sorted copy of int slice s using bubble sort:`
In your case, you working on `s` reference

Reply:
If I'm not wrong, slice by default would not be passed by reference, it would be by value so I'm not modifying the original slice passed in. Again if I'm not wrong, slice will always create a new copy whenever you do anything to it.

I did a quick check on it:

https://play.golang.org/p/r-apnquHx0n